### PR TITLE
Ft add fitbit support 154953733

### PR DESCRIPTION
**Pull request by Jak-Sidious**
#### What does this PR do?

This pr enables the functionality of Fitbit support into the Wger Dashboard.

#### Description of Task to be completed?

Currently, no Fitbit support is provided and as such the user has to manually enter their weight data  

#### How should this be manually tested?

1. Log in as a user, or administrator, 
2. Reach the dashboard
3. Click button for Fitbit login and be routed to the login for a personal Fitbit account
4. Have data displayed on the dashboard

#### Any background context you want to provide?

Previously no support existed for the Fitbit and as such users were relegated to mainly checking  and logging their weight

#### What are the relevant pivotal tracker stories?

#154953733

#### Screenshots (if appropriate)
Login page

Dashboard View

Fitbit Login page

Dashboard with retrieved Fitbit data

### CODE REVIEW

File: wger/core/views/fitbit.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    # App settings from fitbit as regards the app
+    CLIENT_ID = os.getenv('CLIENT_ID')
+    CLIENT_SECRET = os.getenv('CLIENT_SECRET')
+    SCOPE = os.getenv('SCOPE')
+    REDIRECT_URI = os.getenv('REDIRECT_URI')
+    # Authorization and authentication URIs
+    AUTHORIZE_URI = os.getenv('AUTHORIZE_URI')
+    TOKEN_REQUEST_URI = os.getenv("TOKEN_REQUEST_URI")
+
+    def ComposeAuthorizationuri(self):
```
Comment (sir3n-sn) [inline wger/core/views/fitbit.py]: CompositeAuthorization`U`ri not uri

File: wger/core/views/fitbit.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        token['refresh_token'] = response['refresh_token']
+
+        return token
+
+    def RefreshToken(self, token):
+        """ Refresh expired access token """
+        # authentication header
+        # client_id = self.CLIENT_ID.encode('utf-8')
+        # secret = self.CLIENT_SECRET.encode('utf-8')
+        headers = {
```
Comment (sir3n-sn) [inline wger/core/views/fitbit.py]: Remove these comments if you do not need them

File: wger/core/views/misc.py [hunk 3]
```diff
@@ -142,6 +143,35 @@ def dashboard(request):
     return render(request, 'index.html', template_data)
 
 
+@login_required
+def fitbitLogin(request):
+    fitbit = FitBit()
+    login_url = fitbit.ComposeAuthorizationuri()
+    return redirect(login_url)
```
Comment (sir3n-sn) [inline wger/core/views/misc.py]: ComposeAuthorizationuri() use ComposeAuthorizationUri()

File: wger/core/views/misc.py [hunk 4]
```diff
@@ -142,6 +143,35 @@ def dashboard(request):
...
+            weight_entry.user = request.user
+            weight_entry.weight = weight['weight']
+            weight_entry.date = weight['date']
+            try:
+                weight_entry.save()
+            except Exception as e:
+                pass
+    except Exception as e:
+        pass
+    return HttpResponseRedirect(reverse('core:dashboard'))
```
Comment (sir3n-sn) [inline wger/core/views/misc.py]: python pep20 errors should never pass silently 
You could print the error to the cli or log it somewhere

File: wger/core/views/misc.py [hunk 5]
```diff
@@ -142,6 +143,35 @@ def dashboard(request):
...
+            weight_entry = WeightEntry()
+            weight_entry.user = request.user
+            weight_entry.weight = weight['weight']
+            weight_entry.date = weight['date']
+            try:
+                weight_entry.save()
+            except Exception as e:
+                print(e)
+    except Exception as e:
+        print(e)
```
Comment (collin5) [inline wger/core/views/misc.py]: Can you please remove this `print` statement

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (collin5) [inline .vscode/settings.json]: Remove `.vscode` from diff

File: wger/core/views/fitbit.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
+import requests
+import urllib
+import base64
+import os
+
+
+class FitBit:
+    """Class to handle all fitbit operation"""
+    # App settings from fitbit as regards the app
+    CLIENT_ID = os.getenv('CLIENT_ID')
```
Comment (collin5) [inline wger/core/views/fitbit.py]: Let's handle cases where the `env` variable is not set. ie `CLIENT_ID = os.getenv('CLIENT_ID', 'DEFAULT_VALUE')`. Do this for the rest of the env variables.

File: wger/core/views/fitbit.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    CLIENT_SECRET = os.getenv('CLIENT_SECRET')
+    SCOPE = os.getenv('SCOPE')
+    REDIRECT_URI = os.getenv('REDIRECT_URI')
+    # Authorization and authentication URIs
+    AUTHORIZE_URI = os.getenv('AUTHORIZE_URI')
+    TOKEN_REQUEST_URI = os.getenv("TOKEN_REQUEST_URI")
+
+    def ComposeAuthorizationUri(self):
```
Comment (collin5) [inline wger/core/views/fitbit.py]: Let's be consistent and use `camelCase` for the method names. Do this for the rest of the methods.

**Pull request merged**